lib/fog/compute/google/models/server.rb

```diff
@@ -425,10 +425,9 @@ def save(username: nil, public_key: nil)
 
           options = attributes.reject { |_, v| v.nil? }
 
-          if service_accounts && service_accounts[:scopes]
-            options[:service_accounts] = service_accounts.merge(
-              :scopes => map_scopes(service_accounts[:scopes])
-            )
+          if service_accounts && service_accounts[0]
+            service_accounts[0][:scopes] = map_scopes(service_accounts[0][:scopes])
+            options[:service_accounts] = service_accounts
```

Comment:
Since people are possibly already passing in a scopes hash, could we still support that? Maybe modify the request to make sure only one service account is passed in and its in the right format?

Reply:
I will test this tonight and see if it works. But from what I recall it seemed like setting the service accounts and scopes was completely broken on the master branch.

I'll dig into it and let you know.

Reply:
I will not get to this until this weekend. Sorry for the delay.

Reply:
I tried out the latest code from master to see how it behaves. My initial test seems to indicate that simply setting scopes like this does not actually set the service account scopes. When I go into the Google Cloud Console no service account scopes are set on the GCE instance.

```
 {:scopes => ["storage/full"] }
```

Some screen shots from my `pry` session trying to see how this currently works.

![screenshot from 2018-02-26 00-02-25](https://user-images.githubusercontent.com/685221/36655725-1508edac-1a8a-11e8-9578-b4cd34d7f8e5.png)

![screenshot from 2018-02-26 00-03-51](https://user-images.githubusercontent.com/685221/36655737-24a8112a-1a8a-11e8-931a-99debecc497f.png)

Reply:
I think actually I misread this, what I was asking doesn't make sense. I was thinking you were throwing away an option a user was passing in, which is not what is happening here. This actually looks great and I'm going to merge.